python_ta/transforms/type_inference_visitor.py

```diff
@@ -252,16 +258,18 @@ def register_type_constraints_setter():
     return type_visitor
 
 
-def _set_locals_environment(node):
-    node.type_environment = Environment(locals_={name: TYPE_CONSTRAINTS.fresh_tvar() for name in node.locals})
+def _set_environment(node):
+    node.type_environment = Environment(locals_={name: TYPE_CONSTRAINTS.fresh_tvar() for name in node.locals},
+                                        globals_ ={name: TYPE_CONSTRAINTS.fresh_tvar() for name in node.globals})
     if isinstance(node, astroid.FunctionDef):
         node.type_environment.locals['return'] = TYPE_CONSTRAINTS.fresh_tvar()
+        node.type_environment.globals['return'] = TYPE_CONSTRAINTS.fresh_tvar()
```

Comment:
The `return` key is a special one representing the return type of a function. You should not add it to the global environment.